webpack/assets/javascripts/foreman_hostgroups.js

```diff
@@ -0,0 +1,21 @@
+import $ from 'jquery';
+
+export function checkForUnavailablePuppetclasses() {
+  let unavailableClasses = $('#puppet_klasses #selected_classes .unavailable');
+  let puppetKlassesTab = $('#puppet_klasses');
+  let tab = puppetKlassesTab.closest('form').find('.nav-tabs a[href="#puppet_klasses"]');
+  let warningMessage = __('Some Puppet Classes are unavailable in the selected environment');
+  let warning = `<div class="alert alert-warning" id="puppetclasses_unavaliable_warning">
+      <span class="pficon pficon-warning-triangle-o"></span>
+      ${warningMessage}
+    </span>`;
+
+  if (unavailableClasses.size() > 0) {
```

Comment:
Seems like `unavailableClasses.size() > 0` is true even after you change back from the environment that has the classes, so every time I change environments, a new pficon warning is added